Libs/MRML/Core/vtkMRMLMessageCollection.cxx

```diff
@@ -63,7 +63,8 @@ vtkMRMLMessageCollection
 ::GetNumberOfMessagesOfType(unsigned long messageType) const
 {
   int response = 0;
-  for (int i = 0; i < this->Messages.size(); ++i)
+  const unsigned numberOfMessages = static_cast<unsigned>(this->Messages.size());
+  for (unsigned i = 0; i < numberOfMessages; ++i)
     {
     response += static_cast<int>(GetNthMessageType(i) == messageType);
```

Comment:
Bool should not be casted to int. You can use the ternary operator or a simple `if` to increment by 1.

Reply:
I suggest something like this:

```cpp
  int response = 0;
  for (int i = 0; i < static_cast<int>(this->Messages.size()); ++i)
    {
    if (this->GetNthMessageType(i) == messageType)
      {
      response++;
      }
    }
  return response;
```

Reply:
Will do simple `if`.